server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -122,14 +122,14 @@ public final void executeNextPhase(SearchPhase currentPhase, SearchPhase nextPha
          * at least one successful operation left and if so we move to the next phase. If not we immediately fail the
          * search phase as "all shards failed"*/
         if (successfulOps.get() == 0) { // we have 0 successful results that means we shortcut stuff and return a failure
+            final ShardOperationFailedException[] shardSearchFailures = ExceptionsHelper.groupBy(buildShardFailures());
+            Throwable cause = shardSearchFailures.length == 0 ? null :
```

Comment:
shall we add the rest of the array as suppressed exceptions?

Reply:
this can be very very long I am not sure it adds much